src/activities/discussions/GroupSectionRestrictionActionsEntity.js

```diff
@@ -41,4 +42,41 @@ export class GroupSectionRestrictionActionsEntity extends Entity {
 		return fields && fields.value;
 
 	}
+	restrictedTopicCollection() {
+		if (!this._entity) return null;
+
+		const subEntity = this._entity.getSubEntityByClass('restricted-topic');
+		if (!subEntity) {
+			return null;
+		}
+		return (subEntity.getSubEntitiesByRel('item') || []).map(entity => {
+			return new RestrictedTopicCollectionEntity(entity);
+		});
+	}
+	canToggleGroupsRestrictedTopic() {
+		if (!this._entity) return null;
+		const subEntity = this._entity.getSubEntityByClass('restricted-topic');
+		return subEntity && subEntity.hasActionByName(Actions.discussions.groupSectionRestrictions.toggleGroupsRestrictedTopic);
+	}
+	_formatToggleGroupsRestrictedTopicAction(toggleGroupIds) {
+		if (!toggleGroupIds) return;
+		if (!this.canToggleGroupsRestrictedTopic()) return;
+		const subEntity = this._entity.getSubEntityByClass('restricted-topic');
+		const action = subEntity.getActionByName(Actions.discussions.groupSectionRestrictions.toggleGroupsRestrictedTopic);
+		const fields = [
+			{ name: 'toggleGroupIds', value: toggleGroupIds },
```

Comment:
we'll want to add multiple fields with name `toggleGroupIds`, one for each id to toggle, instead of passing a single string for these ids. Note in [the controller ](https://github.com/Brightspace/lms/blob/master/le/discussions/D2L.LE.Discussions/Extensibility/HM/Discussions/Controllers/DiscussionsController.GroupRestrictions.cs#L47) this param is an array instead of a comma-separated string. These multiple fields will be mapped to this array.